documents/write_acls.go

```diff
@@ -84,3 +88,39 @@ func newChangedField(p proofs.Property, leaf *proofs.LeafNode, old bool) changed
 	cf.new = v
 	return cf
 }
+
+// initTransitionRules initiates the transition rules for a given CoreDocumentModel.
+// Collaborators are given default edit capability over all fields of the CoreDocument.
+// if the rules are created already, this is a no-op.
+// if collaborators are empty, it is a no-op
+func (cd *CoreDocument) initTransitionRules(collaborators []identity.DID) {
+	if len(cd.Document.Roles) > 0 && len(cd.Document.TransitionRules) > 0 {
+		return
+	}
+	if len(collaborators) < 0 {
+		return
+	}
+	cd.addCollaboratorsToTransitionRules(collaborators)
+}
+
+// addCollaboratorsToTransitionRules adds the given collaborators to a new transition rule which defaults to
+// granting edit capability over all fields of the document.
+func (cd *CoreDocument) addCollaboratorsToTransitionRules(collaborators []identity.DID) {
+	role := cd.addCollaboratorsToNewRole(collaborators)
+	if role == nil {
+		return
+	}
+	cd.addNewTransitionRule(role, coredocumentpb.FieldMatchType_FIELD_MATCH_TYPE_PREFIX, []byte(compactProperties(CDTreePrefix)), coredocumentpb.TransitionAction_TRANSITION_ACTION_EDIT)
```

Comment:
you are only passing CDtree alone. Init should take specific document prefix and add a new transition rule for that as well

Reply:
Else, no one will ever make any changes to the document underneath